### PR TITLE
jupyter-widget: event-handling via transport

### PR DESCRIPTION
**Pull request by ibgreen**
For #3864 etc

### Background
- Refactor recently added generic event handling for pydeck: playground.js uses `Transport.sendJSONMessage` instead of directly accessing the `transport.jupyterModel`.

### Change List

### CODE REVIEW

File: modules/jupyter-widget/src/playground/playground.js [hunk 1]
```diff
@@ -96,7 +93,8 @@ function sendEventViaTransport(transport, event, data) {
   }
   // TODO Remove circular references without converting to a string
   const deckEvent = JSON.parse(JSON.stringify({event, data}, filterJsonValue));
-  transport.jupyterModel.send(deckEvent);
+
+  transport.sendJSONMessage('deck-event', deckEvent);
```
Comment (ibgreen) [inline modules/jupyter-widget/src/playground/playground.js:97]: @ajduberstein Note that this adds an additional level of nesting that needs to be matched on Python side.
- A transport message has `{type, data}`
- A deck.gl event will be

```js
{
  type: 'deck-event',
  data: {
    type: 'click',
    ...
  }
}
```

Comment (ibgreen) [reply modules/jupyter-widget/src/playground/playground.js:97]: We could avoid this nesting but I recommend a naming convention to separate deck events from other back-channel events. Le me update that.

File: modules/jupyter-widget/src/playground/create-deck.js [hunk 2]
```diff
@@ -113,14 +113,14 @@ function createStandaloneFromProvider({
   // Common deck.gl props for all basemaos
   const handlers = handleEvent
     ? {
-        onClick: info => handleEvent('click', info),
-        onHover: info => handleEvent('hover', info),
-        onResize: size => handleEvent('resize', size),
+        onClick: info => handleEvent('deck-click-event', info),
```
Comment (ibgreen) [inline modules/jupyter-widget/src/playground/create-deck.js:116]: Note the new more specific event names as these now go over the generic transport back-channel, potentially shared with other features.

Comment (ajduberstein) [reply modules/jupyter-widget/src/playground/create-deck.js:116]: The string constants here will need to be changed too: https://github.com/visgl/deck.gl/blob/master/bindings/pydeck/pydeck/widget/widget.py#L93

File: modules/jupyter-widget/src/playground/create-deck.js [hunk 3]
```diff
@@ -113,14 +113,14 @@ function createStandaloneFromProvider({
...
+        onHover: info => handleEvent('deck-hover-event', info),
+        onResize: size => handleEvent('deck-resize-event', size),
         onViewStateChange: ({viewState, interactionState, oldViewState}) => {
```
Comment (ajduberstein) [inline modules/jupyter-widget/src/playground/create-deck.js:119]: I realize now that I added the debouncing logic for this [in pydeck](https://github.com/visgl/deck.gl/blob/master/bindings/pydeck/pydeck/widget/debounce.py) but we may want debouncing on the client instead, given that we'll have to implement it in each binding.

**Pull request merged**